### PR TITLE
[Doc] Logging: Add Fluent Bit sidecar and Grafana Loki to Persist Ray logs

### PR DESCRIPTION
**Pull request by win5923**
## Why are these changes needed?
- Add a tutorial on how to set up Fluent Bit sidecar logging with Grafana Loki on Kubernetes. The FluentBit sidecar container redirects logs from `/tmp/ray/session_latest/logs` to Grafana Loki.
- Using [downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) to fetch pod metadata labels. This allows users to configure Fluent Bit to assign logs based on specific labels, such as `ray.io/cluster`
```yaml
env:
  - name: POD_LABELS
    valueFrom:
      fieldRef:
        fieldPath: metadata.labels['ray.io/cluster']
```
- Users can modify the Fluent Bit ConfigMap's `Labels` to associate logs with different RayCluster / RayJob CRs.
```
Labels RayCluster=raycluster-xxxx
Labels RayJob=rayjob-xxxx
```

## Related issue number
https://github.com/ray-project/kuberay/issues/2515

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml [hunk 1]
```diff
@@ -12,8 +12,12 @@ data:
         Path_Key true
         Refresh_Interval 5
     [OUTPUT]
-        Name stdout
+        Name loki
         Match *
+        Host loki-gateway
+        Port 80
+        Labels RayCluster_name=raycluster-complete-logs
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml:19]: Can we use K8s labels (https://docs.fluentbit.io/manual/pipeline/filters/kubernetes) instead to dynamically set up the log label?

File: doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml [hunk 2]
```diff
@@ -12,8 +12,12 @@ data:
         Path_Key true
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml:12]: Can you open a PR to add a RayCluster YAML to the KubeRay repo and reference the YAML in this doc?

Example:
https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/configuring-autoscaling.html#step-3-create-a-raycluster-custom-resource-with-autoscaling-enabled

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 3]
```diff
@@ -42,18 +42,27 @@ the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
...
 ### Configuring log processing
 The first step is to create a ConfigMap with configuration
 for Fluent Bit.
 
-Here is a minimal ConfigMap which tells a Fluent Bit sidecar to
+The following ConfigMap example configures a Fluent Bit sidecar to:
 * Tail Ray logs.
-* Output the logs to the container's stdout.
+* Sends logs to a Grafana Loki endpoint.
+* Assign logs to a specific tenant using the `Tenant_ID` field, enabling multi-tenancy in Grafana Loki.(In this example, it matches the `test` tenant specified in the Grafana Loki datasource configuration.)
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:65]:
```suggestion
* Assign logs to a specific tenant using the `Tenant_ID` field, enabling multi-tenancy in Grafana Loki. In this example, it matches the `test` tenant specified in the Grafana Loki datasource configuration.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 4]
```diff
@@ -112,34 +122,44 @@ for a single-pod RayCluster will a log-processing sidecar.
...
-
 To deploy the configuration described above, deploy the KubeRay Operator if you haven't yet:
 Refer to the {ref}`Getting Started guide <kuberay-operator-deploy>`
 for instructions on this step.
 
 Now, run the following commands to deploy the Fluent Bit ConfigMap and a single-pod RayCluster with
 a Fluent Bit sidecar.
 ```shell
-kubectl apply -f https://raw.githubusercontent.com/ray-project/ray/releases/2.4.0/doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml
+kubectl apply -f https://raw.githubusercontent.com/ray-project/ray/refs/heads/master/doc/source/cluster/kubernetes/configs/ray-cluster.log.yaml
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:132]: delete the YAML in Ray repo and add a YAML in KubeRay repo instead

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 5]
```diff
@@ -112,34 +122,44 @@ for a single-pod RayCluster will a log-processing sidecar.
...
+* Username: "admin"
+* Password: Get the password using the following command:
+
 ```shell
-kubectl get pod | grep raycluster-complete-logs
+kubectl get secret --namespace default grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
 ```
 
-Examine the FluentBit sidecar's STDOUT to see logs for Ray's component processes.
+Finally, use a LogQL query to view logs for a specific RayCluster or RayJob, and filter by the RayCluster_name:
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:150]:
```suggestion
Finally, use a LogQL query to view logs for a specific RayCluster or RayJob, and filter by the `RayCluster_name`:
```

**Review by kevin85421 — APPROVED** [review]
I will polish the doc after this PR is merged

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 6]
```diff
@@ -42,33 +37,42 @@ the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
...
+
 ### Configuring log processing
 The first step is to create a ConfigMap with configuration
 for Fluent Bit.
 
-Here is a minimal ConfigMap which tells a Fluent Bit sidecar to
+The following ConfigMap example configures a Fluent Bit sidecar to:
 * Tail Ray logs.
-* Output the logs to the container's stdout.
+* Sends logs to a Grafana Loki endpoint.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:59]:
```suggestion
* Send logs to a Grafana Loki endpoint.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 7]
```diff
@@ -42,33 +37,42 @@ the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
...
+* Add metadata to the logs for filtering by labels (e.g., `RayCluster`).
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:60]:
```suggestion
* Add metadata to the logs for filtering by labels, for example, `RayCluster`.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 8]
```diff
@@ -42,33 +37,42 @@ the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
...
-- In addition to streaming logs to stdout, you can use an [OUTPUT] clause to export logs to any
-  [storage backend][FluentBitStorage] supported by Fluent Bit.
 - The `Path_Key true` line above ensures that file names are included in the log records
   emitted by Fluent Bit.
 - The `Refresh_Interval 5` line asks Fluent Bit to refresh the list of files
   in the log directory once per 5 seconds, rather than the default 60.
   The reason is that the directory `/tmp/ray/session_latest/logs/` does not exist
   initially (Ray must create it first). Setting the `Refresh_Interval` low allows us to see logs
   in the Fluent Bit container's stdout sooner.
+- The `POD_LABELS` variable used in the `FILTER` section is populated using the [Kubernetes downward API][KubernetesDownwardAPI]. The label is pulled from the pod's metadata label `ray.io/cluster`, which is defined in the Fluent Bit sidecar container's environment.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:74]:
```suggestion
- The [Kubernetes downward API][KubernetesDownwardAPI] populates the `POD_LABELS` variable used in the `FILTER` section. It pulls the label from the pod's metadata label `ray.io/cluster`, which is defined in the Fluent Bit sidecar container's environment.
```

File: doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md [hunk 9]
```diff
@@ -42,33 +37,42 @@ the Ray container's `/tmp/ray` directory to the relevant `hostPath`.
...
+- The `tenant_id` field allows you to assign logs to different tenants. In this example, the logs are sent to the `test` tenant. This configuration can be adjusted to match the tenant ID set up in your Grafana Loki instance, enabling multi-tenancy support in Grafana.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/persist-kuberay-custom-resource-logs.md:75]:
```suggestion
- The `tenant_id` field allows you to assign logs to different tenants. In this example, Fluent Bit sidecar sends the logs to the `test` tenant. You can adjust this configuration to match the tenant ID set up in your Grafana Loki instance, enabling multi-tenancy support in Grafana.
```

**Review by angelinalg — COMMENTED** [review]
Feel free to correct any inaccuracies in my rewrites. The suggestions are mostly removing passive voice.

**Review by angelinalg — APPROVED** [review]
Approving so as to not block.

**Comment by win5923** [comment]
Thanks for @kevin85421 and @angelinalg's reviews. I've completed the style fixes.

**Pull request merged**